src/client/java/teammates/client/scripts/sql/DataMigrationForAccountAndReadNotificationSql.java

```diff
@@ -141,11 +141,11 @@ protected void migrateEntity(teammates.storage.entity.Account oldAccount) {
 
     private void migrateReadNotification(teammates.storage.entity.Account oldAccount,
             teammates.storage.sqlentity.Account newAccount) {
+
+        HibernateUtil.beginTransaction();
         for (Map.Entry<String, Instant> entry : oldAccount.getReadNotifications().entrySet()) {
-            HibernateUtil.beginTransaction();
             UUID notificationId = UUID.fromString(entry.getKey());
             Notification newNotification = HibernateUtil.get(Notification.class, notificationId);
-            HibernateUtil.commitTransaction();
```

Comment:
Will this cause a problem if one read notification migration fail?

Reply:
If the migration does the early escape at `if (newNotification == null) {`? I think probably not, because it's outside of the entire loop